2023-04-12T00:25:35.527214+0700 ERROR An error has been caught in function 'dispatch_request', process 'MainProcess' (24812), thread 'Thread-2 (process_request_thread)' (8244):
Traceback (most recent call last):

  File "C:\Python310\lib\threading.py", line 966, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x0000018B9DD7C820>
    └ <Thread(Thread-2 (process_request_thread), started daemon 8244)>

  File "C:\Python310\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x0000018B9DD7C550>
    └ <Thread(Thread-2 (process_request_thread), started daemon 8244)>

  File "C:\Python310\lib\threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-2 (process_request_thread), started daemon 8244)>
    │    │        │    └ (<socket.socket fd=380, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=...
    │    │        └ <Thread(Thread-2 (process_request_thread), started daemon 8244)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>>
    └ <Thread(Thread-2 (process_request_thread), started daemon 8244)>

  File "C:\Python310\lib\socketserver.py", line 683, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 43847)
    │    │              └ <socket.socket fd=380, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=(...
    │    └ <function BaseServer.finish_request at 0x0000018B9DEA3E20>
    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>

  File "C:\Python310\lib\socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>
    │    │                   │        └ ('127.0.0.1', 43847)
    │    │                   └ <socket.socket fd=380, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=(...
    │    └ <class 'werkzeug.serving.WSGIRequestHandler'>
    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>

  File "C:\Python310\lib\socketserver.py", line 747, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x0000018B9E523520>
    └ <werkzeug.serving.WSGIRequestHandler object at 0x0000018BA011E7A0>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 361, in handle
    super().handle()

  File "C:\Python310\lib\http\server.py", line 425, in handle
    self.handle_one_request()
    │    └ <function BaseHTTPRequestHandler.handle_one_request at 0x0000018B9E49BE20>
    └ <werkzeug.serving.WSGIRequestHandler object at 0x0000018BA011E7A0>

  File "C:\Python310\lib\http\server.py", line 413, in handle_one_request
    method()
    └ <bound method WSGIRequestHandler.run_wsgi of <werkzeug.serving.WSGIRequestHandler object at 0x0000018BA011E7A0>>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 333, in run_wsgi
    execute(self.server.app)
    │       │    │      └ <werkzeug.debug.DebuggedApplication object at 0x0000018BA00E5B10>
    │       │    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>
    │       └ <werkzeug.serving.WSGIRequestHandler object at 0x0000018BA011E7A0>
    └ <function WSGIRequestHandler.run_wsgi.<locals>.execute at 0x0000018BA012BAC0>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 322, in execute
    for data in application_iter:
                └ <generator object DebuggedApplication.debug_application at 0x0000018BA014C740>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\debug\__init__.py", line 329, in debug_application
    app_iter = self.app(environ, start_response)
               │    │   │        └ <function WSGIRequestHandler.run_wsgi.<locals>.start_response at 0x0000018BA012BA30>
               │    │   └ {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': <_io.BufferedReader name=380>, 'wsgi.errors': <_io.TextIOWr...
               │    └ <Flask 'run'>
               └ <werkzeug.debug.DebuggedApplication object at 0x0000018BA00E5B10>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 2551, in __call__
    return self.wsgi_app(environ, start_response)
           │    │        │        └ <function WSGIRequestHandler.run_wsgi.<locals>.start_response at 0x0000018BA012BA30>
           │    │        └ {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': <_io.BufferedReader name=380>, 'wsgi.errors': <_io.TextIOWr...
           │    └ <function Flask.wsgi_app at 0x0000018B9E8D3D90>
           └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
               │    └ <function Flask.full_dispatch_request at 0x0000018B9E8D3250>
               └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         │    └ <function Flask.dispatch_request at 0x0000018B9E8D31C0>
         └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           │    │           │    │              │    │            └ {}
           │    │           │    │              │    └ 'usersroute'
           │    │           │    │              └ <Rule '/api/users' (DELETE, GET, PUT, HEAD, POST, OPTIONS) -> usersroute>
           │    │           │    └ {'static': <function Flask.__init__.<locals>.<lambda> at 0x0000018B9FEE69E0>, 'usersroute': <function View.as_view.<locals>.v...
           │    │           └ <Flask 'run'>
           │    └ <function Flask.ensure_sync at 0x0000018B9E8D3520>
           └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_restful\__init__.py", line 467, in wrapper
    resp = resource(*args, **kwargs)
           │         │       └ {}
           │         └ ()
           └ <function View.as_view.<locals>.view at 0x0000018BA00DEE60>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           │                       │    │                   └ {}
           │                       │    └ <function Resource.dispatch_request at 0x0000018B9E8EB130>
           │                       └ <app.routes.users.UsersRoute object at 0x0000018BA011ECE0>
           └ <Flask 'run'>

> File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_restful\__init__.py", line 582, in dispatch_request
    resp = meth(*args, **kwargs)
           │     │       └ {}
           │     └ ()
           └ <bound method UsersRoute.post of <app.routes.users.UsersRoute object at 0x0000018BA011ECE0>>

  File "D:\JOBS\pacmann\ujian\backend\app\routes\users.py", line 33, in post
    username = request.form['username']
               └ <Request 'http://127.0.0.1:5000/api/users' [POST]>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\datastructures.py", line 375, in __getitem__
    raise exceptions.BadRequestKeyError(key)
          │          │                  └ 'username'
          │          └ <class 'werkzeug.exceptions.BadRequestKeyError'>
          └ <module 'werkzeug.exceptions' from 'D:\\JOBS\\pacmann\\ujian\\backend\\venv\\lib\\site-packages\\werkzeug\\exceptions.py'>

werkzeug.exceptions.BadRequestKeyError: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.
2023-04-12T00:26:24.725062+0700 ERROR An error has been caught in function 'dispatch_request', process 'MainProcess' (24812), thread 'Thread-3 (process_request_thread)' (25404):
Traceback (most recent call last):

  File "C:\Python310\lib\threading.py", line 966, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x0000018B9DD7C820>
    └ <Thread(Thread-3 (process_request_thread), started daemon 25404)>

  File "C:\Python310\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x0000018B9DD7C550>
    └ <Thread(Thread-3 (process_request_thread), started daemon 25404)>

  File "C:\Python310\lib\threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-3 (process_request_thread), started daemon 25404)>
    │    │        │    └ (<socket.socket fd=912, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=...
    │    │        └ <Thread(Thread-3 (process_request_thread), started daemon 25404)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>>
    └ <Thread(Thread-3 (process_request_thread), started daemon 25404)>

  File "C:\Python310\lib\socketserver.py", line 683, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 43966)
    │    │              └ <socket.socket fd=912, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=(...
    │    └ <function BaseServer.finish_request at 0x0000018B9DEA3E20>
    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>

  File "C:\Python310\lib\socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>
    │    │                   │        └ ('127.0.0.1', 43966)
    │    │                   └ <socket.socket fd=912, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=(...
    │    └ <class 'werkzeug.serving.WSGIRequestHandler'>
    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>

  File "C:\Python310\lib\socketserver.py", line 747, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x0000018B9E523520>
    └ <werkzeug.serving.WSGIRequestHandler object at 0x0000018BA011E7A0>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 361, in handle
    super().handle()

  File "C:\Python310\lib\http\server.py", line 425, in handle
    self.handle_one_request()
    │    └ <function BaseHTTPRequestHandler.handle_one_request at 0x0000018B9E49BE20>
    └ <werkzeug.serving.WSGIRequestHandler object at 0x0000018BA011E7A0>

  File "C:\Python310\lib\http\server.py", line 413, in handle_one_request
    method()
    └ <bound method WSGIRequestHandler.run_wsgi of <werkzeug.serving.WSGIRequestHandler object at 0x0000018BA011E7A0>>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 333, in run_wsgi
    execute(self.server.app)
    │       │    │      └ <werkzeug.debug.DebuggedApplication object at 0x0000018BA00E5B10>
    │       │    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000018BA011D390>
    │       └ <werkzeug.serving.WSGIRequestHandler object at 0x0000018BA011E7A0>
    └ <function WSGIRequestHandler.run_wsgi.<locals>.execute at 0x0000018BA012BAC0>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 322, in execute
    for data in application_iter:
                └ <generator object DebuggedApplication.debug_application at 0x0000018BA0308660>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\debug\__init__.py", line 329, in debug_application
    app_iter = self.app(environ, start_response)
               │    │   │        └ <function WSGIRequestHandler.run_wsgi.<locals>.start_response at 0x0000018BA012BA30>
               │    │   └ {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': <_io.BufferedReader name=912>, 'wsgi.errors': <_io.TextIOWr...
               │    └ <Flask 'run'>
               └ <werkzeug.debug.DebuggedApplication object at 0x0000018BA00E5B10>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 2551, in __call__
    return self.wsgi_app(environ, start_response)
           │    │        │        └ <function WSGIRequestHandler.run_wsgi.<locals>.start_response at 0x0000018BA012BA30>
           │    │        └ {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': <_io.BufferedReader name=912>, 'wsgi.errors': <_io.TextIOWr...
           │    └ <function Flask.wsgi_app at 0x0000018B9E8D3D90>
           └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
               │    └ <function Flask.full_dispatch_request at 0x0000018B9E8D3250>
               └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         │    └ <function Flask.dispatch_request at 0x0000018B9E8D31C0>
         └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           │    │           │    │              │    │            └ {}
           │    │           │    │              │    └ 'usersroute'
           │    │           │    │              └ <Rule '/api/users' (DELETE, GET, PUT, HEAD, POST, OPTIONS) -> usersroute>
           │    │           │    └ {'static': <function Flask.__init__.<locals>.<lambda> at 0x0000018B9FEE69E0>, 'usersroute': <function View.as_view.<locals>.v...
           │    │           └ <Flask 'run'>
           │    └ <function Flask.ensure_sync at 0x0000018B9E8D3520>
           └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_restful\__init__.py", line 467, in wrapper
    resp = resource(*args, **kwargs)
           │         │       └ {}
           │         └ ()
           └ <function View.as_view.<locals>.view at 0x0000018BA00DEE60>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           │                       │    │                   └ {}
           │                       │    └ <function Resource.dispatch_request at 0x0000018B9E8EB130>
           │                       └ <app.routes.users.UsersRoute object at 0x0000018BA011F190>
           └ <Flask 'run'>

> File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_restful\__init__.py", line 582, in dispatch_request
    resp = meth(*args, **kwargs)
           │     │       └ {}
           │     └ ()
           └ <bound method UsersRoute.post of <app.routes.users.UsersRoute object at 0x0000018BA011F190>>

  File "D:\JOBS\pacmann\ujian\backend\app\routes\users.py", line 33, in post
    username = request.form['username']
               └ <Request 'http://127.0.0.1:5000/api/users' [POST]>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\datastructures.py", line 375, in __getitem__
    raise exceptions.BadRequestKeyError(key)
          │          │                  └ 'username'
          │          └ <class 'werkzeug.exceptions.BadRequestKeyError'>
          └ <module 'werkzeug.exceptions' from 'D:\\JOBS\\pacmann\\ujian\\backend\\venv\\lib\\site-packages\\werkzeug\\exceptions.py'>

werkzeug.exceptions.BadRequestKeyError: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.
